src/util/CancellationHandle.h

```diff
@@ -6,17 +6,48 @@
 #define QLEVER_CANCELLATIONHANDLE_H
 
 #include <absl/strings/str_cat.h>
+#include <gtest/gtest_prod.h>
 
 #include <atomic>
 #include <type_traits>
 
 #include "util/CompilerExtensions.h"
 #include "util/Exception.h"
+#include "util/Log.h"
+#include "util/ParseableDuration.h"
 #include "util/TypeTraits.h"
+#include "util/jthread.h"
 
 namespace ad_utility {
 /// Enum to represent possible states of cancellation
-enum class CancellationState { NOT_CANCELLED, MANUAL, TIMEOUT };
+enum class CancellationState {
+  NOT_CANCELLED,
+  WAITING_FOR_CHECK,
+  CHECK_WINDOW_MISSED,
+  MANUAL,
+  TIMEOUT
+};
+
+namespace detail {
+
+constexpr std::chrono::milliseconds DESIRED_CHECK_INTERVAL{50};
```

Comment:
This should be registered more centralized (e.g. in the Constants, or even as one of the `RuntimeParameters()`.

Reply:
Yes, maybe a simple global constant is enough.